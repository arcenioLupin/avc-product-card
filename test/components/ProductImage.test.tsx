import React from "react";
import renderer from 'react-test-renderer';
import ProductCard, { ProductImage } from "../../src/components";
import { product2 } from "../data/products";


describe('ProductImage', () => {
    test('debe mostrar el componente con una imagen personalizada', () => { 
        const wrapper = renderer.create(
            <ProductImage img="hola-mundo.jpg"/>
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
     });

     test('debe mostrar el componente  con la imagen del producto', () => { 
        const wrapper = renderer.create(
            <ProductCard product={product2}>
                {
                    () => (
                        <ProductImage img={product2.img}/>
                    )
                }
            </ProductCard>
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
     })
})