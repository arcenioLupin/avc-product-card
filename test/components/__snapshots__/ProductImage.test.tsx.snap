// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe mostrar el componente  con la imagen del producto 1`] = `
<div
  className="productCard undefined undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductImage debe mostrar el componente con una imagen personalizada 1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="hola-mundo.jpg"
/>
`;
